catroid/src/main/java/org/catrobat/catroid/utils/Utils.java

```diff
@@ -479,6 +479,15 @@ public static double round(double value, int precision) {
 		return (double) Math.round(value * scale) / scale;
 	}
 
+	private static void showDialog(Context context, final Dialog dialog) {
+		((Activity) context).runOnUiThread(new Runnable() {
+			@Override
+			public void run() {
+				dialog.show();
+			}
+		});
+	}
+
```

Comment:
Maybe we should do
```java
if (context instanceof Activity) {
  ((Activity) context).runOnUiThread(…);
} else {
  dialog.show();
}
```

* If the context is an activity, we can call `runOnUiThread`. `runOnUiThread` will either call the code immediately, if the current thread is the UiThread, or postpone it. (this fixes the crash)
* If the context isn't an activity but we're currently executing the code on the UiThread, we can call `dialog.show()`
* If the context isn't an activity and we are not on the UiThread, an exception will be thrown.

This is currently just a dirty, non-oo quick-fix.